irohad/ametsuchi/impl/storage_impl.cpp

```diff
@@ -195,58 +197,26 @@ namespace iroha {
           log_manager_->getChild("MutableStorageImpl"));
     }
 
-    void StorageImpl::reset() {
-      resetWsv().match(
-          [this](auto &&v) {
-            log_->debug("drop blocks from disk");
-            block_store_->clear();
-          },
-          [this](auto &&e) {
-            log_->warn("Failed to drop WSV. Reason: {}", e.error);
-          });
-    }
-
-    expected::Result<void, std::string> StorageImpl::resetWsv() {
-      log_->debug("drop wsv records from db tables");
-      try {
-        soci::session sql(*connection_);
-        // rollback possible prepared transaction
-        tryRollback(sql);
-        return PgConnectionInit::resetWsv(sql);
-      } catch (std::exception &e) {
-        return expected::makeError(e.what());
-      }
-    }
-
     void StorageImpl::resetPeers() {
       log_->info("Remove everything from peers table");
       soci::session sql(*connection_);
       expected::resultToOptionalError(PgConnectionInit::resetPeers(sql)) |
           [this](const auto &e) { this->log_->error("{}", e); };
     }
 
-    void StorageImpl::dropStorage() {
-      log_->info("drop storage");
-      if (connection_ == nullptr) {
-        log_->warn("Tried to drop storage without active connection");
-        return;
-      }
-
-      std::unique_lock<std::shared_timed_mutex> lock(drop_mutex_);
-
-      // erase blocks
-      log_->info("drop block store");
+    expected::Result<void, std::string> StorageImpl::dropBlockStorage() {
+      log_->info("drop block storage");
       block_store_->clear();
+      return iroha::expected::Value<void>{};
+    }
 
-      freeConnections();
-      log_->info("Drop database {}", postgres_options_->workingDbName());
-      if (auto e = expected::resultToOptionalError(
-              PgConnectionInit::dropWorkingDatabase(*postgres_options_))) {
-        log_->warn(e.value());
-      }
+    boost::optional<std::shared_ptr<const iroha::LedgerState>>
+    StorageImpl::getLedgerState() const {
+      return ledger_state_;
```

Comment:
It seems no reason to use 'optional' together with 'shared_ptr', because 'shared_ptr' already contains object validity information.

Reply:
this is an implication of a global policy to avoid implicit unset values. so, if a function may or may not return an object, the result must be an explicit optional.

Reply:
> std::optional instead of boost?

this will require a lot of changes in other components that still use boost. better to do it in a separate refactoring PR, like #498